c/indexer.c

```diff
@@ -42,12 +42,30 @@
 
 #include "indexer.h"
 
+#include <endian.h>
+#include <stdbool.h>
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
 
 #define CHUNK (1 << 14) // file input buffer size
 
+offset_t store_offset(offset_t source) {
+    return htobe64(source);
+}
+
+offset_t retrieve_offset(offset_t source) {
+    return be64toh(source);
+}
+
+int32_t store_size(int32_t source) {
+    return htobe32(source);
+}
+
+int32_t retrieve_size(int32_t source) {
+    return be32toh(source);
+}
+
```

Comment:
These are not storing or retrieving anything. They should not have *store* or *retrieve* in their names.
Suggest `encode_offset`, `decode_offset`, `encode_int32`, `decode_int32`

Reply:
Updated the names and also changed the encoding to little-endian.